docs/development/data-flow/README.md

```diff
@@ -0,0 +1,98 @@
+# Data flow and Usage
+
+Regardless of the type of installation of OpenProject, the following diagram provides a high-level overview of through which systems data related to OpenProject is flowing.
+
+
+
+```mermaid
+%%{init: {'theme':'neutral'}}%%
+
+flowchart TD
+  A[Client Browser] -->|"HTTP(s) requests"| B(Load Balancer / Proxy)
+  A1[API or Native clients] -->|"HTTP(s) requests"| B(Load Balancer / Proxy)
+  B -->|Proxy| openproject
+
+  subgraph openproject[OpenProject]
+    C[Puma app server]
+    D[Background worker]
+  end
+    
+
+  subgraph integrations[External integrations]
+	  O[Other integrations]
+    N[Nextcloud]
+  end
+  
+  subgraph services[Services]
+  	M[memcached]
+	  P[PostgreSQL]
+	  S[Object storage or NFS]
+  end
+  
+
+  openproject <--> services
+  openproject --> integrations
+  B <--> integrations
+	
+```
+
+As a web application, the primary data flow is between the Client browser (or attached API clients) through an external proxying web server (a load balancer or proxying server). We're assuming it is the one responsible for terminating TLS connections for the course of this document - although encrypted connections between Load balancer and Puma server are possible.
+
+The external web server acts as a proxy/reverse-proxy for the OpenProject Puma application server, relaying requests for it to handle and respond. In the course of the request, access to external services such as the PostgreSQL database, a caching server, or attached storages might be performed.
+
+In the course of using the application, background tasks are enqueued in the database such as outgoing emails, cleanup tasks, or notification processing. These tasks are performed in a separate process, the background worker queue. This process accesses the same services as the application server process to access or modify data. It might connect to external integrations such as a Nextcloud instance to set up file sharings depending on actions performed by the users.
+
+
+
+## Schema information
+
+OpenProject's database schema is considered an internal API, please do not rely on it as a stable point of references. Modifications to the database schema are considered a third-party customization and might prevent us from giving proper support.
+
+We recommend you use existing database tools to inspect and export the database as an ERD. That being said, we've prepared an export of an ERD of the current state of the database. Newer additions to it might be missing.
```

Comment:
I'd reference the version the current diagram describes and that we don't plan to update it regularly. 
Since you write "Newer additions to it might be missing" people might expect the schema to always be upwards compatible so I would stress that the schema might change (adding & removing) between patch level releases.